agent/consul/controller/reconciler.go

```diff
@@ -0,0 +1,51 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/consul/acl"
+)
+
+// Request contains the information necessary to reconcile a config entry.
+// This includes only the information required to uniquely identify the
+// config entry.
+type Request struct {
+	Kind string
+	Name string
+	Meta *acl.EnterpriseMeta
+}
+
+// RequeueAfterError is an error that allows a Reconciler to override the
+// exponential backoff behavior of the Controller, rather than applying
+// the backoff algorithm, returning a RequeueAfterError will cause the
+// Controller to reschedule the Request at a given time in the future.
+type RequeueAfterError time.Duration
```

Comment:
Does it make sense to include a cause on this error?

Reply:
I don't think so, in this particular case this is just meant as a mechanism for controlling the explicit retry semantics of the controller -- unless we add in a bunch of internal logging around queueing/retries then a cause won't really ever get used. I personally am ok with just delegating those sorts of logging bits to someone implementing a reconciler -- just throw a logging statement before explicitly returning a `RequeueAfterError`.